packages/rpc-spec/src/rpc-api.ts

```diff
@@ -43,17 +43,14 @@ export function createRpcApi<TRpcMethods extends RpcApiMethods>(config?: RpcApiC
                     TRpcMethods[TMethodName] extends CallableFunction ? TRpcMethods[TMethodName] : never
                 >
             ): PendingRpcApiRequest<ReturnType<TRpcMethods[TMethodName]>> {
-                const params = config?.parametersTransformer
-                    ? config?.parametersTransformer(rawParams, methodName)
-                    : rawParams;
-                const responseTransformer = config?.responseTransformer
-                    ? config?.responseTransformer<ReturnType<TRpcMethods[TMethodName]>>
-                    : (rawResponse: unknown) => rawResponse as ReturnType<TRpcMethods[TMethodName]>;
-                return {
-                    methodName,
-                    params,
-                    responseTransformer,
-                };
+                const rawRequest = Object.freeze({ methodName, params: rawParams });
```

Comment:
Ooh, good idea and also would be smart to have a `isFrozenObject()` test for this.